client/app/bundles/course/assessment/components/FileManager/Toolbar.tsx

```diff
@@ -0,0 +1,85 @@
+import { CSSProperties, ChangeEventHandler } from 'react';
+import { injectIntl, WrappedComponentProps } from 'react-intl';
+import { Button, Grid } from '@mui/material';
+import { Add as AddIcon, Delete as DeleteIcon } from '@mui/icons-material';
+
+import t from './translations.intl';
+
+/**
+ * Types are all any for now because DataTable is not fully typed.
+ */
+interface ToolbarProps extends WrappedComponentProps {
+  selectedRows;
+  onAddFiles: (files: File[]) => void;
+  onDeleteFileWithRowIndex: (index: number) => void;
+}
+
+const styles: { [key: string]: CSSProperties } = {
+  uploadInput: {
+    cursor: 'pointer',
+    position: 'absolute',
+    top: 0,
+    bottom: 0,
+    right: 0,
+    left: 0,
+    opacity: 0,
+  },
+};
+
+const Toolbar = (props: ToolbarProps): JSX.Element => {
+  const { intl } = props;
+
+  const handleDeleteFiles = (e): void => {
+    e.preventDefault();
+
+    props.selectedRows?.data?.forEach((row) => {
+      props.onDeleteFileWithRowIndex?.(row.dataIndex);
+    });
+  };
+
+  const handleFileInputChange: ChangeEventHandler<HTMLInputElement> = (e) => {
+    e.preventDefault();
+
+    const input = e.target;
+    const files = input.files;
+    if (!files) return;
+
+    props.onAddFiles?.(Array.from(files));
+
+    input.value = '';
```

Comment:
Because [according to the documentation, `value` attribute for `<input type="file">` is of type string, _not a nullable string_](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#value).

Previously, in `MaterialUploader`, this attribute was set to `null` 'so that the files can be selected again.' This type mismatch was not caught then because `MaterialUploader` was written in JavaScript.